### ISSUE TITLE
Support using AWS Instance Profile instead of access keys

### ISSUE THREAD

**Issue by beanaroo** [ToolJet/ToolJet#1737]
### Is your proposal related to a problem?

When hosting ToolJet in AWS, it may be preferable to make use of the available Instance Profile for credentials instead of setting up dedicated IAM access keys.

### Describe the solution you'd like

Have Access Key and Secret Key optional when creating a datasource for DynamoDB or S3

### Describe alternatives you've considered

None

### If the feature is approved, would you be willing to submit a PR?

I would be willing to attempt the required server changes. I believe the relevant section for DynamoDB is here: https://github.com/ToolJet/ToolJet/blob/26c9cc655c9774936e1007c07263cac89cdf181d/server/plugins/datasources/dynamodb/index.ts#L52-L53

Basically, if access keys are not provided, use [EC2MetadataCredentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EC2MetadataCredentials.html) instead.

I am not confident enough to attempt any frontend changes

**Comment by akshaysasidrn** [comment]
Hey @beanaroo, sure go-ahead to create a PR for implementing this.

For the frontend, I think the changes would be to add some additional fields to support this and for that, you would only have to tweak this [schema](https://github.com/ToolJet/ToolJet/blob/develop/frontend/src/Editor/DataSourceManager/SourceComponents/Database/Dynamodb.schema.json) and UI components will be generated accordingly.

**Comment by arpitnath** [comment]
resolved #5506
